### PR TITLE
overlord/state: add Change.Err

### PR DESCRIPTION
**Pull request by niemeyer**

### CODE REVIEW

File: overlord/state/change_test.go [hunk 1]
```diff
@@ -223,3 +161,60 @@ func (cs *changeSuite) TestState(c *C) {
...
+		"- (Download|Activate) \\((Download|Activate) error\\)")
+}
+
+func (cs *changeSuite) TestNeedsLock(c *C) {
+	st := state.New(nil)
+	st.Lock()
+	chg := st.NewChange("install", "...")
+	st.Unlock()
+
+	funcs := []func(){
```
Comment (mvo5) [inline overlord/state/change_test.go]: Nice!

File: overlord/state/change_test.go [hunk 2]
```diff
@@ -223,3 +161,60 @@ func (cs *changeSuite) TestState(c *C) {
...
+	t1 := st.NewTask("download", "Download")
+	t2 := st.NewTask("activate", "Activate")
+
+	chg.AddTask(t1)
+	chg.AddTask(t2)
+
+	c.Assert(chg.Err(), IsNil)
+
+	t1.SetStatus(state.ErrorStatus)
+	c.Assert(chg.Err(), IsNil)
```
Comment (mvo5) [inline overlord/state/change_test.go]: I tripped over this when reading the test, i.e. I was wondering why t1 in ErrorState is giving chg.Err() == nil and t2 != nil. Maybe a small comment like `// t2 still running` (or not in an end-state yet) or something to give a hint to the reader?

**Comment by mvo5** [comment]
This looks good. One tiny suggestion but not a blocker.

**Comment by mvo5** [comment]
retest this please

File: overlord/state/change_test.go (hunk 2 above)
Comment (niemeyer) [inline overlord/state/change_test.go]: Good idea.

File: overlord/state/change.go [hunk 3]
```diff
@@ -211,6 +214,51 @@ func (c *Change) SetStatus(s Status) {
...
+	var buf bytes.Buffer
+	buf.WriteString("cannot perform the following tasks:\n")
+	for _, te := range e.errors {
+		fmt.Fprintf(&buf, "- %s (%s)\n", te.task, te.error)
+	}
+	return strings.TrimSuffix(buf.String(), "\n")
+}
+
+// Err returns an error value based on errors that were logged for tasks registered
+// in this change, or nil if the task is not in ErrorStatus.
```
Comment (pedronis) [inline overlord/state/change.go]: "nil if the change is not in ErrorStatus"? (first if)

Comment (niemeyer) [inline overlord/state/change.go]: Oops, thanks.

**Pull request merged**